### PR TITLE
Update controller runtime

### PR DESCRIPTION
**Pull request by Mojachieee**
Updates the controller runtime to 0.8, which is what we use in the storageos operator

### CODE REVIEW

File: controllers/etcdcluster_controller_test.go [hunk 1]
```diff
@@ -267,8 +266,23 @@ func (s *controllerSuite) testClusterController(t *testing.T) {
 			err = try.Eventually(func() error {
 				var actual v1.PersistentVolumeClaimList
 				err := s.k8sClient.List(s.ctx, &actual, client.InNamespace(etcdCluster.Namespace))
+
+				// Since we can't control when PVCs with the pvc-protection are deleted
+				// we will remove them from this list
+				var pvcsWOProtectionFinalizer v1.PersistentVolumeClaimList
```
Comment (Mojachieee) [inline controllers/etcdcluster_controller_test.go:272]: It's probably worth having an extra close look at this, potentially we can sove this some other way

File: controllers/etcdpeer_controller_test.go [hunk 2]
```diff
@@ -279,7 +277,17 @@ func (s *controllerSuite) testPeerController(t *testing.T) {
 		t.Log("The PVC is deleted")
 		err = s.k8sClient.Get(s.ctx, peerKey, &actualPvc)
 		require.NoError(t, client.IgnoreNotFound(err), "unexpected error")
-		assert.Error(t, err, "expected a NotFound error for deleted PVC")
+
+		// Either the PVC gets deleted or it's held up by the pvc-protection finalizer
```
Comment (Mojachieee) [inline controllers/etcdpeer_controller_test.go:281]: Same point here: it's worth taking a detailing look

File: controllers/etcdcluster_controller_test.go [hunk 3]
```diff
@@ -267,8 +266,23 @@ func (s *controllerSuite) testClusterController(t *testing.T) {
...
+				pvcsWOProtectionFinalizer.TypeMeta = actual.TypeMeta
+				pvcsWOProtectionFinalizer.ListMeta = actual.ListMeta
+				for _, pvc := range actual.Items {
+					// If a PVC doesn't have a deletion timestamp or it has some
+					// other finalizer we shouldn't consider it deleted
+					if pvc.DeletionTimestamp != nil && len(pvc.Finalizers) == 1 && pvc.Finalizers[0] == "kubernetes.io/pvc-protection" {
```
Comment (mhmxs) [inline controllers/etcdcluster_controller_test.go:278]: I don't know all the details, but [0] looks a bit odd for me. Should a proper find solution make this expression more robust?

File: controllers/etcdcluster_controller_test.go (hunk 1 above)
Comment (mhmxs) [reply controllers/etcdcluster_controller_test.go:272]: Could you please describe the main issue?

**Review by mhmxs — APPROVED** [review]

File: controllers/etcdcluster_controller_test.go (hunk 3 above)
Comment (Mojachieee) [reply controllers/etcdcluster_controller_test.go:279]: Discussed offline

**Pull request merged**